tests/functional/test_clidriver.py

```diff
@@ -103,6 +103,21 @@ def test_imds_region_is_used_as_fallback_with_v2_support(self):
         self.driver.main(['ec2', 'describe-instances'])
         self.assertEqual(capture.region, 'us-mars-2')
 
+    def test_session_user_agent_extra(self):
+        # if "distro" failed during getting linux version we'll have
+        # just "linux" in the user agent which shouldn't happen
+        self.assertFalse("linux" in self.session.user_agent_extra)
```

Comment:
For this test, I would prefer if we just use the unit test to catch this as we can control when the exception is thrown. While we don't expect the exception to be thrown, it would be weird if a user was running the test on a machine that did throw the exception and now they cannot get the tests to pass.

Also if the distribution id had `linux` in the name, it would also cause the test to fail which also something we would not want.